### PR TITLE
feat(macros): Propagate params handle_reply, handle_signal from grpogram to async_main

### PR DESCRIPTION
**Pull request by vobradovich**
Resolves # https://github.com/gear-tech/sails/issues/309.

### CODE REVIEW

File: macros/core/src/program/mod.rs [hunk 1]
```diff
@@ -12,19 +13,33 @@ use syn::{
...
-    gen_gprogram_impl(program_impl)
+    let args = parse_args(args);
+    gen_gprogram_impl(program_impl, args)
+}
+
+fn parse_args(args: TokenStream2) -> ProgramArgs {
+    syn::parse2(args).unwrap_or_else(|err| {
+        abort!(
+            err.span(),
+            "`gprogram` attribute can only contain `handle_reply` and `handle_signal` parameters: {}",
```
Comment (DennisInSky) [inline macros/core/src/program/mod.rs:39]: How do you know here what went wrong? To my POV, this error should be raise in place where it has appropriate context

File: macros/core/src/program/mod.rs [hunk 2]
```diff
@@ -332,8 +349,21 @@ fn generate_handle<'a>(
...
 
+    let mut args = Vec::with_capacity(2);
+    if let Some(handle_reply) = program_args.handle_reply() {
+        args.push(quote!(handle_reply = #handle_reply));
+    }
+    if let Some(handle_signal) = program_args.handle_signal() {
+        args.push(quote!(handle_signal = #handle_signal));
+    }
+    let async_main_args = if args.is_empty() {
+        quote!()
```
Comment (DennisInSky) [inline macros/core/src/program/mod.rs:360]: Doesn't make much sense, the below quote will result in nothing if `args` is empty

File: macros/core/src/program/args.rs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+            }
+
+            match &*name {
+                "handle_reply" => {
+                    attrs.handle_reply = Some(path);
+                }
+                "handle_signal" => {
+                    attrs.handle_signal = Some(path);
+                }
+                _ => return Err(syn::Error::new_spanned(name, "unknown parameter")),
```
Comment (DennisInSky) [inline macros/core/src/program/args.rs:46]: I would abort here as this is the place where we know the proper error context

File: macros/core/src/program/mod.rs (hunk 2 above)
Comment (vobradovich) [reply macros/core/src/program/mod.rs:360]: it is for generating `#[gstd::async_main]` without `()`.

Comment (DennisInSky) [reply macros/core/src/program/mod.rs:360]: Does it matter? Can't it have empty braces?

File: macros/core/src/program/args.rs (hunk 3 above)
Comment (vobradovich) [reply macros/core/src/program/args.rs:46]: Spanned error is returned here

File: macros/core/src/program/mod.rs (hunk 1 above)
Comment (vobradovich) [reply macros/core/src/program/mod.rs:39]: Takes span from the error

File: macros/core/src/program/mod.rs (hunk 2 above)
Comment (vobradovich) [reply macros/core/src/program/mod.rs:360]: you can) just to minimize changes in insta files with this PR

**Review by DennisInSky — APPROVED** [review]

**Pull request merged**